src/python/pants/source/source_root.py

```diff
@@ -186,14 +207,20 @@ def get_source_roots(self) -> SourceRoots:
 class SourceRootRequest:
     """Find the source root for the given path."""
 
-    path: str
+    path: PurePath
+
+    def __post_init__(self) -> None:
+        if ".." in str(self.path).split(os.path.sep):
```

Comment:
we should use the pathlib api here and avoid converting to strings and using os.path....

Reply:
There is no pathlib api for splitting a path into segments, as far as I can tell.